qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1492,15 +1492,14 @@ def set_html(self, html, base_url=QUrl()):
         # percent encoded content is 2 megabytes minus 30 bytes.
         self._widget.setHtml(html, base_url)
 
-    def _show_error_page(self, url, error):
+    def _show_error_page(self, url: QUrl, error: str) -> None:
         """Show an error page in the tab."""
         log.misc.debug("Showing error page for {}".format(error))
-        url_string = url.toDisplayString()
-        error_page = jinja.render(
-            'error.html',
-            title="Error loading page: {}".format(url_string),
-            url=url_string, error=error)
-        self.set_html(error_page)
+        error_url = QUrl("qute://error")
+        queries = QUrlQuery()
+        queries.setQueryItems((("u", url.toDisplayString()), ("e", error)))
```

Comment:
Why `toDisplayString()` and not `toString(QUrl.FullyEncoded)`?

Reply:
It's arbitrary, but I find that reasonable because they will (probably? I don't know. QUrl parses URLs in `TolerantMode` by default. May need some tests.) round-trip correctly and `qute_error` can just use the string directly without another `toDisplayString`.